pkgs/by-name/so/sozi/package.nix

```
@@ -0,0 +1,36 @@
+{ appimageTools, lib, fetchurl }:
+let
+  pname = "sozi";
+  version = "23.7.25-1690311612";
+
+  src = fetchurl {
+    url = "https://github.com/sozi-projects/Sozi/releases/download/v23.07/Sozi-${version}.AppImage";
+    hash = "sha256-QHvgevv60ZTkkdt+IWaCuXt0XVnhe5Q5oROwa2LFie8=";
+  };
+
+  appimageContents = appimageTools.extract {
+    inherit version pname src;
+  };
+in appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraPkgs = pkgs: with pkgs; [ polkit udev ];
+
+  extraInstallCommands = ''
+
+    install -m 444 -D ${appimageContents}/sozi.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/sozi.desktop \
+      --replace 'Exec=AppRun' 'Exec=sozi'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
```

I'm not familiar with `appimageTools.wrapType2`, so the following suggestion may not apply well, if `appimageTools.wrapType2` behaves similar to `mkDerviation` I'd like to recommend the following as it seems clearer to me. Maybe someone more familiar with  would like to chime in? 🙂 
```suggestion
  installPhase = ''
    runHook preInstall
    install -m 444 -D ${appimageContents}/sozi.desktop -t $out/share/applications
        cp -r ${appimageContents}/usr/share/icons $out/share
    runHook postInstall
  '';

  postInstall = ''  
    substituteInPlace $out/share/applications/sozi.desktop \
      --replace 'Exec=AppRun' 'Exec=sozi'
  '';
```

---

Please undo this last change. `buildFHSEnv` only accepts the following args: https://nixos.org/manual/nixpkgs/unstable/#sec-fhs-environments

---

Thanks for the clarification on this, @eclairevoyant, and providing a link to the documentation. With my comment I was hoping to get a conversation started before any change was made, it seems I was not able to communicate that well :)